web/base_viewer.js

```diff
@@ -1476,6 +1478,35 @@ class BaseViewer {
     this._setCurrentPageNumber(pageNumber, /* resetCurrentPageView = */ true);
     this.update();
   }
+
+  /**
+   * @type {number} One of the values in {ReadingDirection}.
+   */
+  get direction() {
+    return this._direction;
+  }
+
+  /**
+   * @param {number} dir
+   */
+  set direction(direction) {
+    if (this._direction === direction) {
+      return;
+    }
+    if (!Object.values(ReadingDirection).includes(direction)) {
+      throw new Error(`Invalid document direction: ${direction}`);
+    }
+    this._direction = direction;
+    this.container.classList.toggle(
+      "dir-ltr",
+      ReadingDirection.LTR === direction
+    );
+    this.container.classList.toggle(
+      "dir-rtl",
+      ReadingDirection.RTL === direction
+    );
+    this.eventBus.dispatch("directionchanged", { source: this, direction });
+  }
```

Comment:
Please change this to
```suggestion
    this._readingDirection = direction;
    this.eventBus.dispatch("readingdirectionchanged", { source: this, direction });
    
    this._updateReadingDirection();
  }
```
by removing the `this.container.classList.toggle(...)` lines from here and instead adding them to a new `_updateReadingDirection` helper method placed just below instead.
Then, in https://github.com/mozilla/pdf.js/blob/master/web/pdf_single_page_viewer.js, please add an *empty* `_updateReadingDirection() {}` line at the end of the file.